Spigot-API-Patches/0243-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,331 @@
+@@ -13,6 +13,8 @@ import java.util.Set;
+ import java.util.UUID;
+ import java.util.function.Consumer;
+ import java.util.logging.Logger;
++
++import io.papermc.paper.sound.SoundBuilder;
+ import org.bukkit.Warning.WarningState;
+ import org.bukkit.advancement.Advancement;
+ import org.bukkit.block.data.BlockData;
+@@ -1193,6 +1195,42 @@ public final class Bukkit {
+         return server.createMerchant(title);
+     }
+ 
++    /**
++     * Creates an empty SoundBuilder
++     *
++     * @param sound the type of the sound, which will be played
++     * @return a new sound builder
++     */
++    @NotNull
++    public static SoundBuilder createSoundBuilder(@NotNull Sound sound) {
++        return server.createSoundBuilder(sound);
++    }
++
++    /**
++     * Creates an empty SoundBuilder
++     *
++     * @param sound the type of the sound, which will be played
++     * @param volume the volume of the sound, which will be played
++     * @return a new sound builder
++     */
++    @NotNull
++    public static SoundBuilder createSoundBuilder(@NotNull Sound sound, @NotNull Float volume) {
++        return server.createSoundBuilder(sound, volume);
++    }
++
++    /**
++     * Creates an empty SoundBuilder
++     *
++     * @param sound the type of the sound, which will be played
++     * @param volume the volume of the sound, which will be played
++     * @param pitch the pitch of the sound, which will be played
++     * @return a new sound builder
++     */
++    @NotNull
++    public static SoundBuilder createSoundBuilder(@NotNull Sound sound, @NotNull Float volume, @NotNull Float pitch) {
++        return server.createSoundBuilder(sound, volume, pitch);
++    }
++
+     /**
+      * Gets user-specified limit for number of monsters that can spawn in a
+      * chunk.
+diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
+index 8e01e1bbb7cfd98b47dbdb3a0e132dafd1413de6..772f9741b852417dbd3861b2e6bd646a3ae2fb03 100644
+--- a/src/main/java/org/bukkit/Server.java
++++ b/src/main/java/org/bukkit/Server.java
+@@ -13,6 +13,8 @@ import java.util.Set;
+ import java.util.UUID;
+ import java.util.function.Consumer;
+ import java.util.logging.Logger;
++
++import io.papermc.paper.sound.SoundBuilder;
+ import org.bukkit.Warning.WarningState;
+ import org.bukkit.advancement.Advancement;
+ import org.bukkit.block.data.BlockData;
+@@ -1002,6 +1004,24 @@ public interface Server extends PluginMessageRecipient {
+     @NotNull
+     Merchant createMerchant(@Nullable String title);
+ 
++    /**
++     * Creates an empty SoundBuilder.
++     */
++    @NotNull
++    SoundBuilder createSoundBuilder(@NotNull Sound sound);
++
++    /**
++     * Creates an empty SoundBuilder.
++     */
++    @NotNull
++    SoundBuilder createSoundBuilder(@NotNull Sound sound, Float volume);
++
++    /**
++     * Creates an empty SoundBuilder.
++     */
++    @NotNull
++    SoundBuilder createSoundBuilder(@NotNull Sound sound, Float volume, Float pitch);
```

Comment:
Switch to primitive. It makes no sense defining a wrapper class as non-null when there is is primitive type which cannot be null.